src/test/java/uk/nhs/hee/tis/trainee/sync/event/PostEventListenerTest.java

```diff
@@ -82,10 +86,44 @@ void shouldSendRelatedPlacementsToQueueAfterSaveWhenRelatedPlacements() {
     AfterSaveEvent<Post> event = new AfterSaveEvent<>(post, null, null);
     listener.onAfterSave(event);
 
-    verify(fifoMessagingService).sendMessageToFifoQueue(PLACEMENT_QUEUE_URL, placement1);
+    ArgumentCaptor<String> deduplicationIdCaptor1 = ArgumentCaptor.forClass(String.class);
+    verify(fifoMessagingService).sendMessageToFifoQueue(
+        eq(PLACEMENT_QUEUE_URL), eq(placement1), deduplicationIdCaptor1.capture());
     assertThat("Unexpected table operation.", placement1.getOperation(), is(Operation.LOAD));
 
-    verify(fifoMessagingService).sendMessageToFifoQueue(PLACEMENT_QUEUE_URL, placement2);
+    ArgumentCaptor<String> deduplicationIdCaptor2 = ArgumentCaptor.forClass(String.class);
+    verify(fifoMessagingService).sendMessageToFifoQueue(
+        eq(PLACEMENT_QUEUE_URL), eq(placement2), deduplicationIdCaptor2.capture());
     assertThat("Unexpected table operation.", placement2.getOperation(), is(Operation.LOAD));
+
+    assertThat("Unexpected deduplication values.",
+        deduplicationIdCaptor1.getValue().equals(deduplicationIdCaptor2.getValue()), is(false));
+  }
+
+  @Test
+  void shouldUseDifferentDeduplicationIdsForSamePlacementWhenSendingToQueueAfterSave() {
```

Comment:
It feels a bit verbose to include this sort of test for all the listeners that create a deduplication id, which possibly suggests that the creation of the deduplication id should be a single function.